### PR TITLE
chore(GridLayout.Root): migrate to @vanilla-extract/css

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Migrates `GridLayout.Root` to vanilla extract.

## Justify why it's needed

First step into migrating `GridLayout` to `@vanilla-extract/css`

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
* **#4999** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4999?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4998** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4998?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/ForeverPage/ForeverPage.tsx [hunk 1]
```diff
@@ -41,8 +41,8 @@ export const ForeverPage = ({ code: initialCode }: Props) => {
   }
 
   return (
-    <div className={layout}>
-      <GridLayout.Root as="main">
+    <main className={layout}>
+      <GridLayout.Root>
```
Comment (guilhermespopolin) [inline apps/store/src/components/ForeverPage/ForeverPage.tsx:45]: Didn't think it was worth to make `GridLayout.Root` polymorphic for just one use case.

File: apps/store/src/components/GridLayout/GridLayout.tsx [hunk 2]
```diff
@@ -3,25 +3,19 @@
...
-
-  width: '100%',
-  maxWidth: MAX_WIDTH,
-  marginInline: 'auto',
-
-  [mq.lg]: {
-    paddingInline: !nested ? theme.space.lg : 0,
-  },
-}))
+export function Root({
```
Comment (alebedev) [inline apps/store/src/components/GridLayout/GridLayout.tsx:12]: I wonder, would it be a better API to have `gridRoot()` as classname generator?  Probably not worth it for old code, but maybe for new one

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/GridLayout/GridLayout.tsx (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/components/GridLayout/GridLayout.tsx:12]: That could work, yes. Not sure how if it would work for `GridLayout.Content` though 🤔

**Pull request merged**